content/en/docs/tasks/administer-cluster/topology-manager.md

```diff
@@ -222,17 +222,26 @@ You will still have to enable each option using the `TopologyManagerPolicyOption
 
 The following policy options exists:
 * `prefer-closest-numa-nodes` (beta, visible by default; `TopologyManagerPolicyOptions` and `TopologyManagerPolicyBetaOptions` feature gates have to be enabled).
-The `prefer-closest-numa-nodes` policy option is beta in Kubernetes {{< skew currentVersion >}}.
-
-If the `prefer-closest-numa-nodes` policy option is specified, the `best-effort` and `restricted`
-policies will favor sets of NUMA nodes with shorter distance between them when making admission decisions.
-You can enable this option by adding `prefer-closest-numa-nodes=true` to the Topology Manager policy options.
-By default, without this option, Topology Manager aligns resources on either a single NUMA node or
-the minimum number of NUMA nodes (in cases where more than one NUMA node is required). However,
-the `TopologyManager` is not aware of NUMA distances and does not take them into account when making admission decisions.
-This limitation surfaces in multi-socket, as well as single-socket multi NUMA systems,
-and can cause significant performance degradation in latency-critical execution and high-throughput applications if the
-Topology Manager decides to align resources on non-adjacent NUMA nodes.
+  The `prefer-closest-numa-nodes` policy option is beta in Kubernetes {{< skew currentVersion >}}.
+
+  If the `prefer-closest-numa-nodes` policy option is specified, the `best-effort` and `restricted`
+  policies will favor sets of NUMA nodes with shorter distance between them when making admission decisions.
+  You can enable this option by adding `prefer-closest-numa-nodes=true` to the Topology Manager policy options.
+  By default, without this option, Topology Manager aligns resources on either a single NUMA node or
+  the minimum number of NUMA nodes (in cases where more than one NUMA node is required). However,
+  the `TopologyManager` is not aware of NUMA distances and does not take them into account when making admission decisions.
+  This limitation surfaces in multi-socket, as well as single-socket multi NUMA systems,
+  and can cause significant performance degradation in latency-critical execution and high-throughput applications if the
+  Topology Manager decides to align resources on non-adjacent NUMA nodes.
+
+* `max-allowable-numa-nodes` (beta; `TopologyManagerPolicyOptions` and `TopologyManagerPolicyBetaOptions` feature gates have to be enabled)
+  The `max-allowable-numa-nodes` policy option is beta in Kubernetes {{< skew currentVersion >}}.
+
+  The value of `max-allowable-numa-nodes` does not (in and of itself) affect the latency of pod admission. With the `TopologyManager` enabled,
+  the time to admit a pod is tied to the number of NUMA nodes on the physical machine. With the new `max-allowable-numa-nodes` option, nodes
+  with more than 8 NUMA nodes can be alowed to run with the `TopologyManager` enabled. However, it is unknown exactly how much this will slow
```

Comment:
alowed -> allowed

Does this mean there is a soft threshold for `max-allowable-numa-nodes`?

Reply:
TL;DR: we don't know yet.

the gist of this option is that the limit was added in the first place when machines with > 8 NUMA nodes were very rare, so the cap was set "high enough" because there wasn't enough data about how the topology manager algorithm scale.

Nowadays machines with > 8 NUMA nodes are common, and the kubelet just errors out in this scenario (https://github.com/kubernetes/kubernetes/issues/124144 and https://github.com/kubernetes/kubernetes/blob/release-1.30/pkg/kubelet/cm/topologymanager/topology_manager.go#L155 - this is in the setup stage)

So: we don't know how much the admission will slow down with more NUMA nodes to deal with. There's no enough data. Users are now enabled to at least try instead of getting a runtime error (and kubelet crash).